src/python/pants/backend/python/subsystems/pex_bootstrap/_ptex_launcher.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# This bootstrapping technique for lazily loading requirements comes from the pex project test case:
+# https://github.com/pantsbuild/pex/commit/66aa5ef69e8b65004a548b44edc0b81dfa8ec736#diff-b5b57c9d4fc663f59ecaedb851c13bd8R23
+
+import json
+import os
+import sys
+
+from pex import resolver
+from pex.common import open_zip
+from pex.fetcher import Fetcher, PyPIFetcher
+from pex.interpreter import PythonInterpreter
+from pex.pex_builder import PEXBuilder
+from pex.pex_info import PexInfo
+from pex.platforms import Platform
+
+
+self = sys.argv[0]
+filename_base, ext = tuple(os.path.splitext(self))
+hydrated_ipex_file = '{}.ipex'.format(filename_base)
+
+if not os.path.isfile(hydrated_ipex_file):
+  print('Hydrating {} to {}'.format(self, hydrated_ipex_file))
```

Comment:
Probably not a good idea. The underlying pex may have meaningful stdout that the runner is relying upon being un-sullied. STDERR is probably OK though.